sympy/combinatorics/rewritingsystem.py

```diff
@@ -135,11 +163,8 @@ def _remove_redundancies(self, changes=False):
             if v != r:
                 del self.rules[r]
                 removed.add(r)
-                if v != w:
-                    new = self.add_rule(v, w)
-                    added.update(new)
-            else:
-                self.rules[v] = w
+            new = self.add_rule(v, w)
```

Comment:
Actually, using `add_rule` to make deductions from the reduced rules in place seems to be faster. I'm going to allow it during `make_confluent` (it's only used to tidy up a bit, so it's not important that the system is completely reduced) with a keyword argument but otherwise it shouldn't do that.